src/main/java/org/jenkinsci/plugins/workflow/support/steps/input/InputAction.java

```diff
@@ -62,8 +63,8 @@ private synchronized void loadExecutions() {
                 }
             }
             if (execution != null) {
-                // TODO JENKINS-37154 we would rather not block here
-                for (StepExecution se : execution.getCurrentExecutions(true).get()) {
+                // JENKINS-37154 sometimes we must block here in order to get accurate results
+                for (StepExecution se : execution.getCurrentExecutions(true).get(1, TimeUnit.MINUTES)) {
```

Comment:
> Yes, it should, but I would rather not cause other stuff to break just as a result of poor performance.

It's a tradeoff: long waits from a deadlocks/contention vs. some requests failing.  Compromise: 10 seconds?  It's short enough to not trigger timeouts in most cases (where it's driven by an HTTP request), but long enough that failures shouldn't be that common.  

Optionally, set it to 10 seconds and make a system property that can be tweaked if we hit issues.